lib/gui/pages/main/controllers/flash.js

```diff
@@ -92,7 +92,7 @@ module.exports = function (
       // messages on the first place.
       if (error.code === 'EVALIDATION') {
         FlashErrorModalService.show(messages.error.validation())
-      } else if (error.code === 'EUNPLUGGED') {
+      } else if (error.code === 'ENXIO' || error.code === 'ENOENT' || error.code === 'EBUSY' || error.code === 'UNKNOWN') {
```

Comment:
What worries me about this is that some of these errors codes can happen in other circumstances. The writer process may:

- Throw ENOENT on an unrelated `fs` call
- Throw EBUSY when unmounting
- Throw UNKNOWN on other weird Windows use cases

And in all these cases, the current implementation will think that the drive was unplugged.

Reply:
Like what? At this point the only fs calls going on are writes and/or reads (during verification) – the source has already been opened, and unmount errors don't have `error.code` populated. The only way I can see this come from an unrelated part is if someone deletes the source image while it's flashing.

Interestingly enough, I've just deleted the image while flashing on Mac OS, and it just continues as if nothing happened. And yup, the contents are complete and correct from what I can tell. Windows won't let you delete a file that's being used by a running process, and no clue what Linux does.

Reply:
> Like what? At this point the only fs calls going on are writes and/or reads (during verification) 

That sounds like a risky assumption to make. Things can change that would make this misbehave. Unmounting should already be the case.

> Interestingly enough, I've just deleted the image while flashing on Mac OS, and it just continues as if nothing happened. And yup, the contents are complete and correct from what I can tell. Windows won't let you delete a file that's being used by a running process, and no clue what Linux does.

In UNIX systems, when you delete the file, the kernel schedules its actual deletion to the point where all file descriptors pointing to it are closed.